agnpy/tests/test_proton_synchrotron.py

```diff
@@ -0,0 +1,140 @@
+# test the synchrotron module
+import numpy as np
+import astropy.units as u
+from astropy.constants import m_e, m_p
+from astropy.coordinates import Distance
+import pytest
+from pathlib import Path
+from agnpy.emission_regions import Blob
+from agnpy.spectra import ExpCutoffPowerLaw
+from agnpy.synchrotron import Synchrotron, nu_synch_peak
+from agnpy.synchrotron import ProtonSynchrotron
+from agnpy.utils.math import trapz_loglog
+from .utils import (
+    make_comparison_plot,
+    extract_columns_sample_file,
+    check_deviation,
+    clean_and_make_dir,
+)
+
+agnpy_dir = Path(__file__).parent.parent.parent  # go to the agnpy root
+# where to read sampled files
+data_dir = agnpy_dir / "agnpy/data"
+# where to save figures, clean-up before making the new
+figures_dir = clean_and_make_dir(agnpy_dir, "crosschecks/figures/proton_synchrotron")
+
+
+# Define source parameters
+B = 10 * u.G
+redshift = 0.32
+distPKS = Distance(z=redshift) # Already inside blob definition
+doppler_s = 30
+Gamma_bulk = 16
+R = 1e16 * u.cm #radius of the blob
+
+
+class TestProtonSynchrotron:
+    """Class grouping all tests related to the Proton Synchrotron class."""
+
+    def test_synch_reference_sed(self):
+        """Test agnpy proton synchrotron SED against the SED produced by the same particle distribution from Matteo Cerruti's code."""
+        # reference SED
+        lognu, lognuFnu = np.genfromtxt(f"{data_dir}/reference_seds/cerruti_psynch/test_pss.dat",  dtype = 'float', comments = '#', usecols = (0,4), unpack = True)
```

Comment:
There is a function `extract_columns_sample_file` in `tests.utils`, used in all the other tests (e.g. see `test_synchrotron.py`)

Reply:
Ah, ok, I see, the file has several columns and they are actually log values. SO maybe the function is not so useful in this case.